semgrep/semgrep/core_runner.py

```diff
@@ -294,16 +297,13 @@ def _run_rules(
                                     output_json
                                 )
                             else:
-                                print_error(
-                                    f"unexpected non-json output while invoking semgrep core with {' '.join(cmd)} \n {ex}"
+                                raise SemgrepError(
+                                    f"unexpected non-json output while invoking semgrep-core:\n\t{ex}\n{PLEASE_FILE_ISSUE_TEXT}"
                                 )
-                                print_error_exit(f"\n{PLEASE_FILE_ISSUE_TEXT}")
-                                raise ex  # let our general exception handler take care of this
                         except Exception as e:
-                            print_error(
-                                f"non-zero return code while invoking semgrep with:\n\t{' '.join(cmd)}\n{ex} {e}"
+                            raise SemgrepError(
```

Comment:
Including the `cmd` here will produce non-deterministic results in tests. This is because the command contains a temporary file path pointing to the rule file, which will change every run. We can either find some workaround, or just not include the `cmd` info. Personally I don't find it very useful.

Reply:
Yeah I'm fine with removing cmd. Since really with tmp files you cant actually run that command afterwards anyway.